src/main/scala/circt/Convention.scala

```diff
@@ -0,0 +1,52 @@
+package chisel3.circt
+
+import chisel3.RawModule
+import chisel3.experimental.{annotate, ChiselAnnotation}
+import firrtl.annotations.{ModuleName, SingleTargetAnnotation}
+import firrtl.FirrtlUserException
+
+/** Annotation to force the compiler to use a specified port convention.
+  * May only be applied to modules.
+  */
+case class ConventionAnnotation(target: ModuleName, convention: String) extends SingleTargetAnnotation[ModuleName] {
+  def duplicate(target: ModuleName): ConventionAnnotation =
+    ConventionAnnotation(target, convention)
+}
+
+/** The lowering convention of a module
+  * Controls how a module's ports are lowered to verilog.
+  */
+sealed class Convention(val string: String) {
+  def serialize: String = string
+}
```

Comment:
Given that this is declared as `val string`, the PR can actually skip the `def serialize` and just just `convention.string`. The `val` will create a public member.